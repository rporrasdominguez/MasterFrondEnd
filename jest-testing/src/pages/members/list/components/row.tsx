import * as React from 'react';
import { Member } from '../viewModel';

interface Props {
  member: Member;
}

export const Row: React.StatelessComponent<Props> = (props) => (
  <tr className="item">
    <td>
      <img src={props.member.avatarUrl} style={{ maxWidth: '10rem' }} />
    </td>
    <td>
      <span>{props.member.id}</span>
    </td>
    <td>
      <span>{props.member.name}</span>
    </td>
  </tr>
);
