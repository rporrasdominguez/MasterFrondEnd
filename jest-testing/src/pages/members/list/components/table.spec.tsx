import * as React from 'react';
import { Table } from './table';
import { shallow,mount } from 'enzyme';
import { Header } from '../../../../common/components/panel/components';

describe('pages/members/list/components/body specs', () => {

  it('should render as expected when passing one item', () => {
   
    // Arrange
    const props ={
      title:'Test Header',

      members:[
        {
          id: 1,
          name: 'name test',
          avatarUrl: 'url test',
        }   
      ]
           
    }

    // Act
    const component = mount(<Table {...props}/>);

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('should render as expected when passing more one item', () => {
   
    // Arrange
    const props ={
      title:'Test Header',

      members:[
        {
          id: 1,
          name: 'name test',
          avatarUrl: 'url test',
        } ,
        {
          id: 2,
          name: 'name test 2',
          avatarUrl: 'url test 2',
        } ,

      ]
           
    }

    // Act
    const component = mount(<Table {...props}/>);

    // Assert
    expect(component).toMatchSnapshot();
  });
});