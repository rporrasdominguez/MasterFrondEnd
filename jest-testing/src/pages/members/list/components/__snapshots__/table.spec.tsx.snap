// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/body specs should render as expected when passing more one item 1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test",
        "id": 1,
        "name": "name test",
      },
      Object {
        "avatarUrl": "url test 2",
        "id": 2,
        "name": "name test 2",
      },
    ]
  }
  title="Test Header"
>
  <table
    className="table table-striped"
  >
    <Component>
      <thead>
        <tr>
          <th>
            Avatar
          </th>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
        </tr>
      </thead>
    </Component>
    <Component
      members={
        Array [
          Object {
            "avatarUrl": "url test",
            "id": 1,
            "name": "name test",
          },
          Object {
            "avatarUrl": "url test 2",
            "id": 2,
            "name": "name test 2",
          },
        ]
      }
    >
      <tbody
        className="list-items"
      >
        <Component
          key="1"
          member={
            Object {
              "avatarUrl": "url test",
              "id": 1,
              "name": "name test",
            }
          }
        >
          <tr
            className="item"
          >
            <td>
              <img
                src="url test"
                style={
                  Object {
                    "maxWidth": "10rem",
                  }
                }
              />
            </td>
            <td>
              <span>
                1
              </span>
            </td>
            <td>
              <span>
                name test
              </span>
            </td>
          </tr>
        </Component>
        <Component
          key="2"
          member={
            Object {
              "avatarUrl": "url test 2",
              "id": 2,
              "name": "name test 2",
            }
          }
        >
          <tr
            className="item"
          >
            <td>
              <img
                src="url test 2"
                style={
                  Object {
                    "maxWidth": "10rem",
                  }
                }
              />
            </td>
            <td>
              <span>
                2
              </span>
            </td>
            <td>
              <span>
                name test 2
              </span>
            </td>
          </tr>
        </Component>
      </tbody>
    </Component>
  </table>
</Component>
`;

exports[`pages/members/list/components/body specs should render as expected when passing one item 1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test",
        "id": 1,
        "name": "name test",
      },
    ]
  }
  title="Test Header"
>
  <table
    className="table table-striped"
  >
    <Component>
      <thead>
        <tr>
          <th>
            Avatar
          </th>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
        </tr>
      </thead>
    </Component>
    <Component
      members={
        Array [
          Object {
            "avatarUrl": "url test",
            "id": 1,
            "name": "name test",
          },
        ]
      }
    >
      <tbody
        className="list-items"
      >
        <Component
          key="1"
          member={
            Object {
              "avatarUrl": "url test",
              "id": 1,
              "name": "name test",
            }
          }
        >
          <tr
            className="item"
          >
            <td>
              <img
                src="url test"
                style={
                  Object {
                    "maxWidth": "10rem",
                  }
                }
              />
            </td>
            <td>
              <span>
                1
              </span>
            </td>
            <td>
              <span>
                name test
              </span>
            </td>
          </tr>
        </Component>
      </tbody>
    </Component>
  </table>
</Component>
`;
